### PR TITLE
Bug fix: Refactor Find Functionality

### LINKED ISSUE #325: Refactor find code for PersonDetailContainsKeywordAndExamPredicate
Currently, the code goes through a weird pathway before reaching the PersonDetailContainsKeywordAndExamPredicate creation. 
We should 
1. Rename the names of both predicates to something manageable cos both are ridiculously long. Maybe `FindDetailPredicate` and `FindScorePredicate`
2. Have both predicate extend from a parent class that is accepted by find command. Maybe `FindPredicate`
3. Have FindCommandParser parse the user input and decide which predicate to create based on the user prefix input.
4. Create the FindCommand with the appropriate predicate so theres no need for all the extra code within FindCommand and in the predicates
5. Update the UG with the updated implementation

### PR DESCRIPTION
**Pull request by jayllo-c**
Fixes #325

Currently, the find command holds a predicate and creates another
predicate based on whether exam scores are needed.

However, this may be messy due to the fact that the predicate
objects are created in different places despite serving the same
purpose.

Let's refactor the creation of the predicate objects to ensure
that the objects follow a more logical and organized flow.

In doing so, the code will be both more readable and scalable.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/FindCommand.java [hunk 1]
```diff
@@ -48,39 +50,63 @@ public class FindCommand extends Command {
...
+        this.keyword = keyword;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        Exam selectedExam = model.getSelectedExam().getValue();
-        if (predicate.isExamRequired()) {
-            updateFilteredPersonListByScore(model, selectedExam);
+        logger.info("Executing find command");
```
Comment (Zer0Legion) [inline src/main/java/seedu/address/logic/commands/FindCommand.java:69]: Great use of logging!

**Review by Zer0Legion — COMMENTED** [review]
Great change that reduces clutter in the code base!

File: src/main/java/seedu/address/logic/commands/FindCommand.java (hunk 1 above)
Comment (jayllo-c) [reply src/main/java/seedu/address/logic/commands/FindCommand.java:69]: Thanks! I will continue to incorporate logging throughout my code in the coming PRs

File: src/main/java/seedu/address/logic/parser/FindCommandParser.java [hunk 2]
```diff
@@ -49,14 +47,14 @@ private Prefix extractPrefixForFindCommand(ArgumentMultimap argMultimap) throws
...
-     * Checks if the value of the given ArgumentMultimap is a positive integer
+     * Checks if the value of the given {@code ArgumentMultimap} is a positive decimal number
      * if the prefix is PREFIX_LESSTHAN or PREFIX_GREATERTHAN.
      * @throws ParseException if the user input does not conform to the expected format
      */
     private String extractValidKeyword(ArgumentMultimap argMultimap, Prefix prefix) throws ParseException {
         if (prefix.equals(CliSyntax.PREFIX_LESS_THAN) || prefix.equals(CliSyntax.PREFIX_MORE_THAN)) {
-            if (!argMultimap.getValue(prefix).get().matches("\\d+")) {
-                throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+            if (!Score.isValidScoreString(argMultimap.getValue(prefix).get())) {
```
Comment (Pughal77) [inline src/main/java/seedu/address/logic/parser/FindCommandParser.java:56]: Nice code reuse!

**Review by Pughal77 — COMMENTED** [review]
Nice changes!

**Pull request merged**